src/Data/Version/Compat.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Data.Version.Compat (
+  module Base
+, makeVersion
+) where
+import Data.Version as Base
+
+#if !MIN_VERSION_base(4,7,0)
+import Data.Data
+#endif
+
+#if !MIN_VERSION_base(4,8,0)
+import Prelude.Compat
+
+# if __GLASGOW_HASKELL__ >= 708
```

Comment:
I think it would also be fine to constrain by `base` version here.  But I guess both are fine.